### PR TITLE
feat: Override service-account-name for aws-load-balancer-controller addon

### LINKED ISSUE #333: [Bug]: String length check needed for aws_iam_role.irsa name in module kubernetes-addons
### Welcome to Amazon SSP EKS Accelerator!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/aws-samples/aws-eks-accelerator-for-terraform/issues) and didn't find any.

### Amazon EKS Accelerator Release version

v3.5.0

### What is your environment, configuration and the example used?

Terraform v1.1.7

### What did you do and What did you see instead?

Cluster name was mri-dbg-gen-566646271983 but the underlying module tried to create a name larger than 64 characters causing the terraform apply to fail. Some sort of length check needs to be implemented for name on line 47.

```
Error: expected length of name to be in the range (1 - 64), got mri-dbg-gen-566646271983-eks-aws-load-balancer-controller-sa-irsa

   with module.kubernetes-addons.module.aws_load_balancer_controller[0].module.helm_addon.module.irsa[0].aws_iam_role.irsa[0],
   on .terraform\modules\kubernetes-addons\modules\irsa\main.tf line 47, in resource "aws_iam_role" "irsa":
   47:   name        = format("%s-%s-%s", var.addon_context.eks_cluster_id, trim(var.kubernetes_service_account, "-*"), "irsa")
```

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by nitrocode**
### What does this PR do?

🛑 Please open an issue first to discuss any significant work and flesh out details/direction - we would hate for your time to be wasted.
Consult the [CONTRIBUTING](https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/main/CONTRIBUTING.md#contributing-via-pull-requests) guide for submitting pull-requests.

### Motivation

- Resolves #333 
- Partially resolves https://github.com/aws-ia/terraform-aws-eks-blueprints/issues/1183

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [ ] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [ ] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [x] E2E Test successfully complete before merge?

### Additional Notes

```hcl
  # module.addons.module.aws_load_balancer_controller[0].module.helm_addon.module.irsa[0].aws_iam_role.irsa[0] will be created
  + resource "aws_iam_role" "irsa" {
      ...
      + description           = "AWS IAM Role for the Kubernetes service account aws-lb-sa."
      ...
      + name                  = "<long-eks-cluster-name>-aws-lb-sa-irsa"
      ...
    }
```

### CODE REVIEW

**Comment by nitrocode** [comment]
@askulkarni2 @bryantbiggs  please review soon. This is blocking a clusters aws lb controller irsa and we're trying to avoid having to use a fork in the mean time. 🙏

File: modules/kubernetes-addons/aws-load-balancer-controller/locals.tf [hunk 1]
```diff
@@ -4,13 +4,14 @@ locals {
```
Comment (bryantbiggs) [inline modules/kubernetes-addons/aws-load-balancer-controller/locals.tf:4]: the only change we need to make is on line 3

```hcl
service_account_name = try(var.helm_config.service_account_name, "${local.name}-sa")
```

Comment (nitrocode) [reply modules/kubernetes-addons/aws-load-balancer-controller/locals.tf:4]: @bryantbiggs addressed. Thank you for the review.

**Comment by nitrocode** [comment]
@bryantbiggs All tests pass! Thanks again. I suppose i was complicating it. I'm glad the changes could be reduced to a single line.

Should i put in a pr to do this across all the modules?

**Comment by bryantbiggs** [comment]
@nitrocode - yes please, if you would like to that would be great. thank you

**Comment by nitrocode** [comment]
@bryantbiggs Cool, I'll put in a PR then

**Comment by nitrocode** [comment]
Can we release a patch release or should we wait until my follow-up PR ?

**Comment by bryantbiggs** [comment]
this project doesn't release on each merge to main - a release will be made either later today or early tomorrow

**Pull request merged**